vcd/resource_vcd_vapp_network.go

```diff
@@ -230,11 +230,11 @@ func resourceVappNetworkRead(d *schema.ResourceData, meta interface{}) error {
 	if c := vAppNetwork.Configuration; c != nil {
 		d.Set("fence_mode", c.FenceMode)
 		if c.IPScopes != nil {
-			d.Set("gateway", c.IPScopes.IPScope.Gateway)
-			d.Set("netmask", c.IPScopes.IPScope.Netmask)
-			d.Set("dns1", c.IPScopes.IPScope.DNS1)
-			d.Set("dns2", c.IPScopes.IPScope.DNS2)
-			d.Set("dnsSuffix", c.IPScopes.IPScope.DNSSuffix)
+			d.Set("gateway", c.IPScopes.IPScope[0].Gateway)
+			d.Set("netmask", c.IPScopes.IPScope[0].Netmask)
+			d.Set("dns1", c.IPScopes.IPScope[0].DNS1)
+			d.Set("dns2", c.IPScopes.IPScope[0].DNS2)
+			d.Set("dnsSuffix", c.IPScopes.IPScope[0].DNSSuffix)
```

Comment:
Why have we made this an array, but using only the first occurrence?

Reply:
+1. Seems like there's a need to loop over it and store all the scopes.

Reply:
Guys this different resource - it's vApp. And here is tricky - I am kind'a with idea removing read part which overwrite state and leave with only exist or not. Trick is that configure it as network configuration, but returned as IpScope in array. So direct read with array may create unexpected behaviours. Or it always return only one IpScope - internal stuff of vCD.

Reply:
We discussed this and agreed that @vbauzysvmware will create a separate issue to investigate this part in vApp network as it is out of scope for this PR and needs dedicated investigation.

Reply:
p.s. Please post a link to that issue here when you have it.

Reply:
https://github.com/terraform-providers/terraform-provider-vcd/issues/235